lib/ansible/modules/cloud/amazon/cloudtrail.py

```diff
@@ -19,231 +19,583 @@
-      region: us-east-1
-
-  - name: remove cloudtrail
-    local_action:
-      module: cloudtrail
-      state: disabled
-      name: main
-      region: us-east-1
-"""
-
-HAS_BOTO = False
-try:
-    import boto
-    import boto.cloudtrail
-    from boto.regioninfo import RegionInfo
-    HAS_BOTO = True
-except ImportError:
-    HAS_BOTO = False
+      - A full ARN specifying a valid CloudWatch log group to which CloudTrail logs will be delivered. The log group should already exist.
+      - See U(https://docs.aws.amazon.com/awscloudtrail/latest/userguide/send-cloudtrail-events-to-cloudwatch-logs.html)
+      - "Example arn:aws:logs:us-east-1:123456789012:log-group:CloudTrail/DefaultLogGroup:*"
+      - Required when C(cloudwatch_logs_role_arn)
+    version_added: "2.4"
+  kms_key_id:
+    description:
+      - Specifies the KMS key ID to use to encrypt the logs delivered by CloudTrail. This also has the effect of enabling log file encryption.
+      - The value can be an alias name prefixed by "alias/", a fully specified ARN to an alias, a fully specified ARN to a key, or a globally unique identifier.
+      - Examples
+      - alias/MyAliasName
+      - "arn:aws:kms:us-east-1:123456789012:alias/MyAliasName"
+      - "arn:aws:kms:us-east-1:123456789012:key/12345678-1234-1234-1234-123456789012"
+      - 12345678-1234-1234-1234-123456789012
+      - See U(https://docs.aws.amazon.com/awscloudtrail/latest/userguide/encrypting-cloudtrail-log-files-with-aws-kms.html)
+    version_added: "2.4"
+  tags:
+    description:
+      - A hash/dictionary of tags to be applied to the CloudTrail resource.
+      - Remove completely or specify an empty dictionary to remove all tags.
+    default: {}
+    version_added: "2.4"
+
+extends_documentation_fragment:
+- aws
+- ec2
+'''
+
+EXAMPLES = '''
+- name: create single region cloudtrail
+  cloudtrail:
+    state: present
+    name: default
+    s3_bucket_name: mylogbucket
+    s3_key_prefix: cloudtrail
+    region: us-east-1
+
+- name: create multi-region trail with validation and tags
+  cloudtrail:
+    state: present
+    name: default
+    s3_bucket_name: mylogbucket
+    region: us-east-1
+    is_multi_region_trail: true
+    enable_log_file_validation: true
+    tags:
+      environment: dev
+      Name: default
+
+- name: pause logging the trail we just created
+  cloudtrail:
+    state: present
+    name: default
+    enable_logging: false
+    s3_bucket_name: mylogbucket
+    region: us-east-1
+    is_multi_region_trail: true
+    enable_log_file_validation: true
+    tags:
+      environment: dev
+      Name: default
+
+- name: delete a trail
+  cloudtrail:
+    state: absent
+    name: default
+'''
+
+RETURN = '''
+exists:
+    description: whether the resource exists
+    returned: always
+    type: bool
+    sample: true
+trail:
+    description: CloudTrail resource details
+    returned: always
+    type: complex
+    sample: hash/dictionary of values
+    contains:
+        TrailARN:
```

Comment:
This should be trail_arn.

The camel_dict_to_snake_dict should be used to convert boto3 objects to ansible objects before returning them through module.exit_json